homeassistant/components/spaceapi.py

```diff
@@ -0,0 +1,171 @@
+"""
+Support for the SpaceAPI.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/spaceapi/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_FRIENDLY_NAME, ATTR_ICON, ATTR_LATITUDE,
+    ATTR_LOCATION, ATTR_LONGITUDE, ATTR_STATE, ATTR_UNIT_OF_MEASUREMENT,
+    CONF_ADDRESS, CONF_EMAIL, CONF_ENTITY_ID, CONF_SENSORS, CONF_STATE,
+    CONF_URL)
+import homeassistant.core as ha
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_ADDRESS = 'address'
+ATTR_API = 'api'
+ATTR_CLOSE = 'close'
+ATTR_CONTACT = 'contact'
+ATTR_ISSUE_REPORT_CHANNELS = 'issue_report_channels'
+ATTR_LASTCHANGE = 'lastchange'
+ATTR_LOGO = 'logo'
+ATTR_NAME = 'name'
+ATTR_OPEN = 'open'
+ATTR_SENSORS = 'sensors'
+ATTR_SPACE = 'space'
+ATTR_UNIT = 'unit'
+ATTR_URL = 'url'
+ATTR_VALUE = 'value'
+
+CONF_CONTACT = 'contact'
+CONF_HUMIDITY = 'humidity'
+CONF_ICON_CLOSED = 'icon_closed'
+CONF_ICON_OPEN = 'icon_open'
+CONF_IRC = 'irc'
+CONF_ISSUE_REPORT_CHANNELS = 'issue_report_channels'
+CONF_LOCATION = 'location'
+CONF_LOGO = 'logo'
+CONF_MAILING_LIST = 'mailing_list'
+CONF_PHONE = 'phone'
+CONF_SPACE = 'space'
+CONF_TEMPERATURE = 'temperature'
+CONF_TWITTER = 'twitter'
+
+DATA_SPACEAPI = 'data_spaceapi'
+DEPENDENCIES = ['http']
+DOMAIN = 'spaceapi'
+
+ISSUE_REPORT_CHANNELS = [CONF_EMAIL, CONF_IRC, CONF_MAILING_LIST, CONF_TWITTER]
+
+SENSOR_TYPES = [CONF_HUMIDITY, CONF_TEMPERATURE]
+SPACEAPI_VERSION = 0.13
+
+URL_API_SPACEAPI = '/api/spaceapi'
+
+LOCATION_SCHEMA = vol.Schema({
+    vol.Optional(CONF_ADDRESS): cv.string,
+}, required=True)
+
+CONTACT_SCHEMA = vol.Schema({
+    vol.Optional(CONF_EMAIL): cv.string,
+    vol.Optional(CONF_IRC): cv.string,
+    vol.Optional(CONF_MAILING_LIST): cv.string,
+    vol.Optional(CONF_PHONE): cv.string,
+    vol.Optional(CONF_TWITTER): cv.string,
+}, required=False)
+
+STATE_SCHEMA = vol.Schema({
+    vol.Required(CONF_ENTITY_ID): cv.entity_id,
+    vol.Optional(CONF_ICON_CLOSED): cv.string,
+    vol.Optional(CONF_ICON_OPEN): cv.string,
+}, required=False)
+
+SENSOR_SCHEMA = vol.Schema(
+    {vol.In(SENSOR_TYPES): [cv.entity_id]}
+)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_CONTACT): CONTACT_SCHEMA,
+        vol.Required(CONF_ISSUE_REPORT_CHANNELS):
+            vol.All(cv.ensure_list, [vol.In(ISSUE_REPORT_CHANNELS)]),
+        vol.Required(CONF_LOCATION): LOCATION_SCHEMA,
+        vol.Required(CONF_LOGO): cv.string,
+        vol.Required(CONF_SPACE): cv.string,
+        vol.Required(CONF_STATE): STATE_SCHEMA,
+        vol.Required(CONF_URL): cv.string,
```

Comment:
Voluptuous has a nice URL validator built in, `vol.Url`. #validateAllTheThings :)

Reply:
It would be `cv.url` in our case.